lib/index.js

```diff
@@ -840,8 +840,13 @@ Stream.prototype.destroy = function () {
     _(this._consumers).each(function (consumer) {
         self._removeConsumer(consumer);
     });
+    this._observers.length = 0;
+
     if (this.source) {
-        this.source._removeConsumer(this);
+        var source = this.source;
+        source._removeConsumer(this);
+        source._removeObserver(this);
+        source = null;
```

Comment:
True, it should be gc'd; I just like explicitly setting references to null when I'm done with them as an added hint, as gc can be wonky sometimes.

Old habits. I'll push an update.